doc/source/whatsnew/v3.0.0.rst

```diff
@@ -645,6 +645,7 @@ Other
 - Bug in :meth:`DataFrame.apply` where passing ``engine="numba"`` ignored ``args`` passed to the applied function (:issue:`58712`)
 - Bug in :meth:`DataFrame.eval` and :meth:`DataFrame.query` which caused an exception when using NumPy attributes via ``@`` notation, e.g., ``df.eval("@np.floor(a)")``. (:issue:`58041`)
 - Bug in :meth:`DataFrame.eval` and :meth:`DataFrame.query` which did not allow to use ``tan`` function. (:issue:`55091`)
+- Bug in :meth:`DataFrame.query` which raised a ``KeyError`` when the expression contained column names with characters like ``#``. (:issue:`59285`)
```

Comment:
Can you also add a note for your fix for 49633?

Reply:
I added it to the same line. Lmk if that should be a separate line!